File: app/src/main/java/com/grupo1/deremate/MainActivity.java
package com.grupo1.deremate;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import android.os.Bundle;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.grupo1.deremate.fragments.DeliversFragment;
import com.grupo1.deremate.fragments.HomeFragment;
import com.grupo1.deremate.fragments.ProfileFragment;
import com.grupo1.deremate.fragments.ScannerFragment;
import com.grupo1.deremate.fragments.SettingsFragment;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        BottomNavigationView bottomNavigationView = findViewById(R.id.bottomNavigation);
        bottomNavigationView.setSelectedItemId(R.id.home);

        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.fragment_container, new HomeFragment())
                    .commit();
        }

        bottomNavigationView.setOnItemSelectedListener(item -> {
            Fragment selectedFragment = null;
            if (item.getItemId() == R.id.home) {
                selectedFragment = new HomeFragment();
            } else if (item.getItemId() == R.id.delivers) {
                selectedFragment = new DeliversFragment();
            } else if (item.getItemId() == R.id.settings) {
                selectedFragment = new SettingsFragment();
            } else if (item.getItemId() == R.id.profile) {
                selectedFragment = new ProfileFragment();
            } else if (item.getItemId() == R.id.scanner) {
                selectedFragment = new ScannerFragment();
            }

            if (selectedFragment != null) {
                getSupportFragmentManager().beginTransaction()
                        .replace(R.id.fragment_container, selectedFragment)
                        .commit();
            }

            return true;
        });
    }
}

File: app/src/main/java/com/grupo1/deremate/MainActivity.kt
package com.grupo1.deremate

import android.content.Intent
import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.color.DynamicColors
import com.grupo1.deremate.databinding.ActivityMainBinding
import com.grupo1.deremate.repository.UserRepository
import com.grupo1.deremate.session.SessionManager
import com.grupo1.deremate.util.BiometricHelper
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@AndroidEntryPoint
class MainActivity : AppCompatActivity() {
    @Inject lateinit var sessionManager: SessionManager
    @Inject lateinit var userRepository: UserRepository

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        DynamicColors.applyToActivityIfAvailable(this)
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()

        val token = sessionManager.getToken()
        if (token != null) {
            val biometricHelper = BiometricHelper(
                activity = this,
                onSuccess = { navigateWithTokenValidation() },
                onFailure = { navigateToLogin() }
            )

            if (biometricHelper.isBiometricAvailable()) {
                biometricHelper.showBiometricPrompt()
            } else {
                navigateWithTokenValidation()
            }
        } else {
            navigateToLogin()
        }
    }

    private fun navigateWithTokenValidation() {
        sessionManager.isValidToken { isValid ->
            runOnUiThread {
                if (isValid) {
                    startActivity(Intent(this, DashboardActivity::class.java))
                } else {
                    navigateToLogin()
                }
                finish()
            }
        }
    }

    private fun navigateToLogin() {
        startActivity(Intent(this, LoginActivity::class.java))
        finish()
    }
}
